python/wpt/grouping_formatter.py

```diff
@@ -182,7 +194,7 @@ def data_was_for_expected_result(data):
             return True
         return "known_intermittent" in data and data["status"] in data["known_intermittent"]
 
-    def test_end(self, data: dict) -> Optional[UnexpectedResult]:
+    def test_end(self, data: dict) -> Union[UnexpectedResult, str, None]:
```

Comment:
```suggestion
    def test_end(self, data: dict) -> Optional[UnexpectedResult] | str:
```

However, neither of these seem to be good solutions to the underlying issue - the base class and derived classes have different return types - `ServoHandler` returns `Optional[UnexpectedResult]` in some cases, but `ServoFormatter` never returns an `UnexpectedResult` by design. @mrobinson do you know if one these should be overridden methods in the first place?

Reply:
Hrm. I think they have to be overridden because they are called by the WPT testing infrastructure. Can we just have `ServoFormatter.test_end` annotated to return `Optional[UnexpectedResult]`, but always return `None`?